phpBB/phpbb/controller/helper.php

```diff
@@ -142,6 +142,12 @@ public function route($route, array $params = array(), $is_amp = true, $session_
 
 		$base_url = $context->getBaseUrl();
 
+		// Regarding PHPBB3-13493 - if we request url ending on / there is no $page_name
+		if (!strpos($base_url, $page_name))
```

Comment:
You are correct. It's taken from there. What it means afaik is: If the last char is a `/` then `$page_name` will be an empty string.